File: iks-core/src/com/iks/cms/core/sql/query/SqlQuery.java
package com.iks.cms.core.sql.query;

import java.util.*;

/**
 * @author Igor Kaynov
 */
public abstract class SqlQuery {
  protected void appendList( StringBuilder sql, List list, String init, String sep, String end ) {
    boolean first = true;
    for( Object s : list ) {
      if( first ) {
        sql.append( init );
      } else {
        sql.append( sep );
      }
      sql.append( s == null ? null : s.toString() );
      first = false;
    }
    if (end != null) {
      sql.append( end );
    }
  }
}

File: iks-core/src/com/iks/cms/core/sql/query/SqlQuery.kt
package com.iks.cms.core.sql.query

import java.util.*

/**
 * @author Igor Kaynov
 */
abstract class SqlQuery {
  protected fun appendList(sql: StringBuilder, list: List<Any?>, init: String, sep: String, end: String?) {
    var first = true
    for (s in list) {
      if (first) {
        sql.append(init)
      } else {
        sql.append(sep)
      }
      sql.append(s?.toString())
      first = false
    }
    if (end != null) {
      sql.append(end)
    }
  }
}
